### PR TITLE
#167733675 Accommodation Booking

### PR DESCRIPTION
**Pull request by Segun-Ogundipe**
#### What does this PR do?
Have an endpoint that creates accommodation booking
#### Description of Task to be completed?
- Implement an accommodation booking model
- Implement a service method for creating a booking in the database
- Implement a controller method for getting a booking object from the request and saving it in the database
- Implement middleware for validating bookings
- Create an accommodation booking route
    - POST `/api/booking/accommodation` for booking accommodations
- Write unit tests for the functionalities mentioned above.
#### How should this be manually tested?
- Clone the repo and cd into the project folder
- Update or copy `.env` file from `.env.sample`
- Checkout to `ft-accommodations-booking-167733675` branch and run `npm install`
- Run `docker-compose up -d` to start docker instances, or `npm run start:dev` 
- Run `npm run migrate` to run database migration (Adds User to the database)
- Test route on Postman with **POST**: **localhost:3000/api/booking/accommodation** example body below
    ```
    {
            "checkIn": "2020-01-01",
            "checkOut": "2020-01-02",
            "userId": 1,
            "roomId": 1
    } 

#### What are the relevant pivotal tracker stories?
[#167733675](https://www.pivotaltracker.com/story/show/167733675)

### CODE REVIEW

**Review by dchima — APPROVED** [review]
The models look good. Great work Segun.

**Review by dchima — APPROVED** [review]

**Review by emmanuelnwankwo — APPROVED** [review]

**Comment by efeguono** [comment]
@Segun-Ogundipe Properly rebase your branch.

File: swagger.json [hunk 1]
```diff
@@ -787,7 +846,131 @@
...
+                },
+                "example": {
+                    "status": "fail",
+                    "error": {
+                        "message": "A server error prevented your request from being processed"
+                    }
+                }
+            }
         }
     }
```
Comment (Baystef) [inline swagger.json:1083]: Great work with the docs and your feature. Please add a newline to eof

**Review by Baystef — APPROVED** [review]

File: .travis.yml [hunk 2]
```diff
@@ -7,7 +7,7 @@ cache:
 services:
   - postgresql
 before_script:
-  - psql -c 'create database jubilee_test;' -U postgres
+  - psql -c 'create database jubilee_ test;' -U postgres
```
Comment (amarachukwu-agbo) [inline .travis.yml]: Why are we changing the database name here?

File: src/test/booking.test.js [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+        roomId
+      };
+
+      const response = await chai
+        .request(server)
+        .post('/api/booking/accommodation')
+        .set('Authorization', `Bearer ${adminToken}`)
+        .send(booking);
+      expect(response).to.have.status(201);
+      expect(response.body.data).to.be.a('object');
```
Comment (amarachukwu-agbo) [inline src/test/booking.test.js]: Can we check the contents of the object to ensure we have the correct object data? Check out the `toMatchObject` method

File: src/test/booking.test.js [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+      };
+
+      const response = await chai
+        .request(server)
+        .post('/api/booking/accommodation')
+        .set('Authorization', `Bearer ${adminToken}`)
+        .send(booking);
+      expect(response).to.have.status(400);
+      expect(response.body.error).to.be.a('object');
+      expect(response.body.error.message).to.be.a('string');
```
Comment (amarachukwu-agbo) [inline src/test/booking.test.js]: Please check that the error message being returned matches what you are expecting

Comment (amarachukwu-agbo) [reply src/test/booking.test.js]: Also implement this in all cases where you have used this same assertion

File: .travis.yml (hunk 2 above)
Comment (Segun-Ogundipe) [reply .travis.yml]: That was an oversight. Have been corrected

File: src/test/booking.test.js (hunk 4 above)
Comment (Segun-Ogundipe) [reply src/test/booking.test.js]: Done

File: src/test/booking.test.js (hunk 3 above)
Comment (Segun-Ogundipe) [reply src/test/booking.test.js]: Couldn't find the toMatchObject method in chai's assertion library

Comment (amarachukwu-agbo) [reply src/test/booking.test.js]: My bad, I assumed the test runner is Jest; that method is for Jest. For Mocha and Chai, you can use `include` or `deep include`

Comment (Segun-Ogundipe) [reply src/test/booking.test.js]: Have been implemented

**Pull request merged**